src/stan/lang/generator/generate_function_instantiation_body.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef STAN_LANG_GENERATOR_GENERATE_FUNCTION_INSTANTIATION_BODY_HPP
+#define STAN_LANG_GENERATOR_GENERATE_FUNCTION_INSTANTIATION_BODY_HPP
+
+#include <stan/lang/ast.hpp>
+#include <stan/lang/generator/generate_function_name.hpp>
+#include <stan/lang/generator/generate_function_instantiation_template_parameters.hpp>
+#include <ostream>
+#include <vector>
+#include <string>
+
+namespace stan {
+  namespace lang {
+
+    /**
+     * Generate the body for a function instantiation (e.g., the 
+     * call to the templated function with all templated arguments based on 
+     * double).
+     * Requires precalculated flags for whether it is an RNG, uses the log
+     * density accumulator or is a probability function, to the
+     * specified stream.
+     *
+     * @param[in] fun function declaration
+     * @param[in] is_rng true if function is an RNG
+     * @param[in] is_lp true if function accesses log density
+     * accumulator
+     * @param[in] is_log true if function is log probability function
+     * @param[in] rng_class class of the RNG being used (required by xxx_rng 
+     * functions)
+     * @param[in,out] o stream for generating
+     */
+    void generate_function_instantiation_body(const function_decl_def& fun,
+                                    bool is_rng, bool is_lp, bool is_log,
+                                    const std::string& rng_class,
+                                    std::ostream& o) {
+      o << "{" << EOL;
+      o << "\t";
+      if (!fun.return_type_.is_void()) {
```

Comment:
replace all `\t` with 2 spaces - generated code should contain spaces not tabs - change this everywhere.